### ISSUE TITLE
BUG [PF Experimental] - Pathfinding has artifacting

### ISSUE THREAD

**Issue by definitely-nobody-is-here** [definitely-nobody-is-here/BattleBoxes_Multiplayer#177]
**Describe the bug**
The A* Pathfinding algorithm combined with the first iteration of accessibility checks creates some odd pathfinding artifacts. Sometimes the bot cannot path to you and other times it attempts to fly through collisions or goes all the way to x = 2 or y = 2. Other times it cannot "see" you at all.

**To Reproduce**
1. Enable test bots in the ["Experimental-PathFind"](https://github.com/definitely-nobody-is-here/BattleBoxes_Multiplayer/tree/Experimental-PathFind) branch.
2. Move around until artifacting appears

**Expected behavior**
Artifacting is a major bug.

**Screenshots**
![image](https://user-images.githubusercontent.com/67865358/125217925-d06df580-e28f-11eb-8d8b-c90c2e31445d.png)
![image](https://user-images.githubusercontent.com/67865358/125217932-d663d680-e28f-11eb-9f01-75726041bc46.png)
![image](https://user-images.githubusercontent.com/67865358/125217950-debc1180-e28f-11eb-86dd-0c13288902d7.png)
![image](https://user-images.githubusercontent.com/67865358/125217956-e085d500-e28f-11eb-9276-6cffe3680495.png)
Image Sources:
[definitely-nobody-is-here](https://github.com/definitely-nobody-is-here)
[The-God-coder](https://github.com/The-God-coder)

**Environment**
 - Browser [e.g. Chrome 90, Safari, Edge]
 - Version v1.5.0-ExperimentalPF-Preview-07/09/2021

**Additional context**
Heuristics or incorrect checks causing it?

**Update 07/12/2021**
There is an overflow problem, may be because the array is in the "yx" order and the original uses the "xy" order.

**Comment by definitely-nobody-is-here** [comment]
It's not the heuristics or the gravity checks causing this problem, it might be the actual algorithm.

**Comment by definitely-nobody-is-here** [comment]
wowowowo

**Comment by definitely-nobody-is-here** [comment]
![image](https://user-images.githubusercontent.com/67865358/125364684-e982af00-e340-11eb-9faa-074bdaacf161.png)

**Comment by definitely-nobody-is-here** [comment]
![image](https://user-images.githubusercontent.com/67865358/125366763-c4903b00-e344-11eb-9555-f003ce6c9411.png)
:/
It doesn't work anymore.

**Comment by definitely-nobody-is-here** [comment]
![image](https://user-images.githubusercontent.com/67865358/125367237-c9a1ba00-e345-11eb-933c-91a891dd7a55.png)
It likes going towards (0, 0) and also straying random directions.

**Comment by definitely-nobody-is-here** [comment]
![image](https://user-images.githubusercontent.com/67865358/125367853-f0142500-e346-11eb-9ea9-362c59fe5490.png)
It appears we have an overflow problem.

**Comment by definitely-nobody-is-here** [comment]
![image](https://user-images.githubusercontent.com/67865358/125369167-c27cab00-e349-11eb-8878-2a51e34346fe.png)
This?

**Comment by definitely-nobody-is-here** [comment]
![image](https://user-images.githubusercontent.com/67865358/125369844-218eef80-e34b-11eb-89c2-7b31e52d9a6f.png)
@The-God-coder says the path was overflowing and causing problems. Nobody knows why or how but it happens at the edges of the path.

**Comment by definitely-nobody-is-here** [comment]
@maitian352 found this while messing around
![image](https://user-images.githubusercontent.com/67865358/125683489-0a448cd6-76f0-4108-9631-25df861e7452.png)
![image](https://user-images.githubusercontent.com/67865358/125683462-ff01a2b8-b5a8-4b67-805e-caced7c96eb7.png)
The x and y are STRINGS (somehow) and adding stuff completely breaks it.

**Comment by definitely-nobody-is-here** [comment]
Cause: the `for` loops outputted strings.

**Comment by definitely-nobody-is-here** [comment]
Cause: The `for` loop outputted strings and caused the neighbor finding to jump. Fixing this eliminated the artifacting of x and y values.

**Comment by maitian352** [comment]
yay bug fixed

**Comment by maitian352** [comment]
try bugnav to fix bug
